extensions/amp-twitter/1.0/test/test-component.js

```diff
@@ -0,0 +1,168 @@
+    const requestResizeSpy = env.sandbox.spy();
+    const wrapper = mount(
+      <Twitter
+        tweetid="1356304203044499462"
+        style={{
+          'width': '500px',
+          'height': '600px',
+        }}
+        requestResize={requestResizeSpy}
+      />
+    );
+
+    const iframe = wrapper.find('iframe').getDOMNode();
+    const sentinel = JSON.parse(iframe.getAttribute('name'))['attributes'][
+      'sentinel'
+    ];
+
+    const mockEvent = new CustomEvent('message');
+    mockEvent.data = serializeMessage('embed-size', sentinel, {
+      'height': 1000,
+    });
+    mockEvent.source = wrapper
+      .getDOMNode()
+      .querySelector('iframe').contentWindow;
+    window.dispatchEvent(mockEvent);
+
+    expect(requestResizeSpy).to.have.been.calledOnce;
+  });
+
+  it('should change height', async () => {
+    const wrapper = mount(
+      <Twitter
+        tweetid="1356304203044499462"
+        style={{
+          'width': '500px',
+          'height': '600px',
+        }}
+      />
+    );
+
+    const iframe = wrapper.find('iframe').getDOMNode();
+    const sentinel = JSON.parse(iframe.getAttribute('name'))['attributes'][
+      'sentinel'
+    ];
+
+    const mockEvent = new CustomEvent('message');
+    mockEvent.data = serializeMessage('embed-size', sentinel, {
+      'height': 1000,
+    });
+    mockEvent.source = wrapper
+      .getDOMNode()
+      .querySelector('iframe').contentWindow;
+    window.dispatchEvent(mockEvent);
+
+    wrapper.update();
+
+    await waitFor(
+      () => wrapper.find('div').at(0).prop('style').height == 1000,
+      'Height changes'
+    );
+
+    expect(wrapper.find('div').at(0).prop('style').height).to.equal(1000);
+  });
+
+  it('should dispatch load event', async () => {
+    const ref = createRef();
+    const onReadyState = env.sandbox.spy();
+    const wrapper = mount(
+      <Twitter
+        ref={ref}
+        tweetid="1356304203044499462"
+        style={{
+          'width': '500px',
+          'height': '600px',
+        }}
+        onReadyState={onReadyState}
+      />
+    );
+
+    let api = ref.current;
+    expect(api.readyState).to.equal('loading');
+    expect(onReadyState).to.not.be.called;
+
+    await wrapper.find('iframe').invoke('onLoad')();
+    api = ref.current;
+    expect(api.readyState).to.equal('complete');
+    expect(onReadyState).to.be.calledOnce.calledWith('complete');
+  });
+
+  it('should reset iframe on pause', () => {
+    const ref = createRef();
+    const wrapper = mount(
+      <WithAmpContext playable={true}>
+        <Twitter
+          ref={ref}
+          tweetid="1356304203044499462"
+          style={{
+            'width': '500px',
+            'height': '600px',
+          }}
+        />
+      </WithAmpContext>
+    );
+    expect(wrapper.find('iframe')).to.have.lengthOf(1);
+
+    const iframe = wrapper.find('iframe').getDOMNode();
+    let iframeSrc = iframe.src;
+    const iframeSrcSetterSpy = env.sandbox.spy();
```

Comment:
Sinon has specific syntax for wrapping a property getter/setter.
Would this work?
```suggestion
    const spy = env.sandbox.spy(iframe, 'src', ["get", "set"]]);
```

Source: "Using a spy to wrap a property getter and setter" at https://sinonjs.org/releases/latest/spies/

Reply:
That did work, though I excluded `"get"` since we only check for the setter to be called. Thanks for the recommendation!

Reply:
woot!